planning/behavior_velocity_planner/src/utilization/util.cpp

```diff
@@ -511,36 +511,21 @@ boost::optional<int64_t> getNearestLaneId(
   const PathWithLaneId & path, const lanelet::LaneletMapPtr lanelet_map,
   const geometry_msgs::msg::Pose & current_pose, boost::optional<size_t> & nearest_segment_idx)
 {
-  boost::optional<int64_t> nearest_lane_id;
-
   nearest_segment_idx = motion_utils::findNearestSegmentIndex(
     path.points, current_pose, std::numeric_limits<double>::max(), M_PI_2);
 
   if (!nearest_segment_idx) {
     return boost::none;
   }
 
-  lanelet::ConstLanelets current_lanes;
-  if (
-    lanelet::utils::query::getCurrentLanelets(
-      lanelet::utils::query::laneletLayer(lanelet_map), current_pose, &current_lanes) &&
-    nearest_segment_idx) {
-    for (const auto & ll : current_lanes) {
-      if (ll.id() == path.points.at(*nearest_segment_idx).lane_ids.at(0)) {
-        nearest_lane_id = ll.id();
-        return nearest_lane_id;
-      }
-    }
+  lanelet::ConstLanelets lanes;
+  for (const auto & point : path.points) {
+    lanes.push_back(lanelet_map->laneletLayer.get(point.lane_ids.at(0)));
```

Comment:
`lane_ids.front()` do not necessarily contain all lane_ids, so  all ids in lane_ids need to be pushed back.